dotCMS/src/main/java/com/dotcms/cube/CubeJSResultSet.java

```diff
@@ -0,0 +1,56 @@
+package com.dotcms.cube;
+
+import com.dotcms.cube.CubeJSResultSet.ResultSetItem;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Represent a Result from RUN a CubeJS Query in a CubeJS Server.
+ */
+public class CubeJSResultSet implements Iterable<ResultSetItem> {
+    private List<ResultSetItem> data;
+    private int size;
```

Comment:
Given that this is stored by actual `data` object. Do we really need `size` attribute?

Reply:
done https://github.com/dotCMS/core/pull/23895/commits/951aeb5323d846f15d0c91e766c156be90af7821#diff-1dadb3f843cc4530acae23891479609d5780ba316cc36d0558c09795d7fecb83R23